File: Java/Armstrong.java
import java.util.Scanner;

public class Armstrong {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int n = sc.nextInt();
        for (int i=0;i<n;i++){
            int temp1 = i;
            int temp2 = i;
            int count = 0;
            int sum =0;
            while(temp1>0){
                temp1/=10;
                count++;
            }
            while(temp2>0){
                int a= temp2%10;
                temp2/=10;
                sum += Math.pow(a,count);}
            if (sum==i){
                System.out.println(sum);
                }

        }
        sc.close();
    }
}

File: C/armstrong.c
#include <stdio.h>
#include <math.h>

int isArmstrong(int num) {
    int originalNum = num;
    int sum = 0;
    int numDigits = 0;
    int digit;

    while (num != 0) {
        num /= 10;
        numDigits++;
    }

    num = originalNum;

    while (num != 0) {
        digit = num % 10;
        sum += pow(digit, numDigits);
        num /= 10;
    }

    return (sum == originalNum);
}

int main() {
    int num;

    printf("Enter a number: ");
    scanf("%d", &num);

    if (isArmstrong(num)) {
        printf("%d is an Armstrong number.\n", num);
    } else {
        printf("%d is not an Armstrong number.\n", num);
    }

    return 0;
}
